src/http_server.cc

```diff
@@ -3078,154 +3078,217 @@ HTTPAPIServer::HandleGenerate(
       req,
       GetModelVersionFromString(model_version_str, &requested_model_version));
 
-  std::string config_json_str = "";
-  RETURN_AND_RESPOND_IF_ERR(
-      req,
-      GetModelConfig(model_name, requested_model_version, &config_json_str));
+  std::map<std::string, triton::common::TritonJson::Value> input_metadata;
+  RETURN_AND_RESPOND_IF_ERR(req, ModelInputMetadata(model_name, requested_model_version,
+      &input_metadata));
 
-  // Parse simple request format
-  triton::common::TritonJson::Value request;
-  RETURN_AND_RESPOND_IF_ERR(req, EVRequestToJson(req, &request));
 
-  // Fallbacks
-  std::string prompt = "";
-  triton::common::TritonJson::Value params_json;
+  // [FIXME] decompression should have been done here. before parsing request body
```

Comment:
Do we need to support compression out of the box? I think it's OK as an enhancement. The simple `curl` use case is pretty minimal as far as I can tell.

Reply:
You are correct that it is just for completeness purposes.